File: misanuncios-ui/src/app/application/services/image.service.ts
import { Injectable } from "@angular/core";
import { ImageRepository } from "../repositories/image.repository";
import { Image } from "../../domain/model/image.model";
import { Observable } from "rxjs";




@Injectable({
  providedIn: 'root',
})
export class ImageService {

  constructor(private imageRepository:ImageRepository) { }

  save(image:string,product_id:number){
    return this.imageRepository.save(image,product_id);
  }
  delete(id:number):Observable<boolean>{
    return this.imageRepository.delete(id);
  }
  findByProduct(idProduct:number):Observable<Image[]>{
    return this.imageRepository.findByProduct(idProduct);
  }





}

File: misanuncios/src/main/java/com/yaagoub/misanuncios/application/service/ImageService.java
package com.yaagoub.misanuncios.application.service;

import com.yaagoub.misanuncios.application.repository.ImageRepository;
import com.yaagoub.misanuncios.domain.Image;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import java.util.List;

@Slf4j
@RequiredArgsConstructor
public class ImageService  {

    private  final  ImageRepository imageRepository;
    public List<Image> getImagesByProduct(long idProduct) {
        return (List<Image>) imageRepository.getImagesByProduct(idProduct);
    }


    public Image save(Image image) {
        return imageRepository.save(image);
    }


    public void delete(Image image) {
         imageRepository.delete(image);
    }
    public  Image findById(long id){
        return imageRepository.findById(id);
    }
}
